lib/service/Controller/Scout/LSPController.dart

```diff
@@ -17,61 +17,63 @@ class LSPController extends GetxController {
   late LongShortPair genericLSP;
   var createAmt = 0.0.obs;
   var redeemAmt = 0.0.obs;
+  var aptAddress = ''.obs;
+  final tokenClient =
+      Web3Client("https://matic-mumbai.chainstacklabs.com", new Client());
   // Hard address listing of all Athletes
 
-  LSPController() {
-    final tokenClient =
-        Web3Client("https://matic-mumbai.chainstacklabs.com", new Client());
-    EthereumAddress address =
-        EthereumAddress.fromHex("0xD3E03e36D70F65A00732F9086D994D83A3EaC286");
-    genericLSP = LongShortPair(address: address, client: tokenClient);
-  }
+  LSPController();
 
   Future<void> mint() async {
+    EthereumAddress address = EthereumAddress.fromHex(aptAddress.value);
+    genericLSP = LongShortPair(address: address, client: tokenClient);
     print("Attempting to MINT LSP");
-    print(await controller.credentials.extractAddress());
     final theCredentials = controller.credentials;
-    BigInt tokensToCreate = BigInt.from(createAmt.value);
-
-    // approve().then((value) async {
-    // });
-    approve().then((value) async {
+    BigInt tokensToCreate =
+        BigInt.from(createAmt.value) * BigInt.from(1000000000000000000);
+    approve(genericLSP, address).then((value) async {
       String txString =
           await genericLSP.create(tokensToCreate, credentials: theCredentials);
       controller.updateTxString(txString); //Sends tx to controller
     });
   }
 
-  Future<void> approve() async {
-    BigInt amount =
-        BigInt.from(createAmt.value) * BigInt.parse('250000000000000000');
+  Future<void> approve(
+      LongShortPair genericLSP, EthereumAddress address) async {
+    print("collaterall amount: ${await genericLSP.collateralPerPair()}");
+    BigInt transferAmount = await genericLSP.collateralPerPair();
+    BigInt amount = BigInt.from(createAmt.value) * transferAmount;
     print("[Console] Inside approve()");
-    EthereumAddress address =
-        EthereumAddress.fromHex("0x76d9a6e4cdefc840a47069b71824ad8ff4819e85");
-    final tokenClient =
-        Web3Client("https://matic-mumbai.chainstacklabs.com", Client());
-    Erc20 axt = Erc20(address: address, client: tokenClient);
-    try {
-      print("[Console] Created a token variable.");
-    } catch (error) {
-      print(error);
-    }
-    print("[Console] Got the amount");
-    EthereumAddress spender =
-        EthereumAddress.fromHex("0xD3E03e36D70F65A00732F9086D994D83A3EaC286");
+    EthereumAddress axtaddress = EthereumAddress.fromHex(AXT.mumbaiAddress);
```

Comment:
I am assuming this is a testnet address?

Reply:
yes, this is a testnet address.